### ISSUE TITLE
[bug] undefined route on prerendering - SvelteKit adapter

### ISSUE THREAD

**Issue by WarningImHack3r** [opral/monorepo#2141]
### Problem

The latest `@inlang/paraglide-js-adapter-sveltekit` version (0.3.0) fails to build ([log here](https://github.com/EmeraldHQ/Website/actions/runs/7702081441/job/20989603622?pr=208)). I tried digging as far as I could but I couldn't find a single breaking change we could be impacted with.

Any idea?

### Expected behavior

The build to not crash

### Reproduction

1. Upgrade to the latest adapter version
2. `vite build`
3. Crash

**Comment by LorisSigrist** [comment]
I'll try to reproduce it, but I already see that you have a bunch of your own utilities for handling the path (`deconstructPathname` etc), so perhaps one of those could be causing something to go wrong. The adapter offers a bunch of built-in functions to deal with the path, that also respect the i18n configuration.

- `i18n.route("/en/translated") => "/untranslated"`
- `i18n.getLanguageFromUrl(url) => "en" | "de" | "..."`
- `i18n.resolveRoute("/untranslated", "en") => "/en/translated"`

**Comment by WarningImHack3r** [comment]
> but I already see that you have a bunch of your own utilities for handling the path (`deconstructPathname` etc)

True, I was focused on building the thing and I didn't really take time to dig into what's available 😅  
I then thought I'm sure there's something for that use later on, but it was done and working so...  

However, thanks for the utilities you just gave me!! I'll replace my code with that.

> perhaps one of those could be causing something to go wrong

It would make sense but I really doubt so: the build _only_ fails with the latest SvelteKit adapter update, and none of the changes you made impact me as far as I could see. It also works fine in dev mode and every single link is correct on the page :/

**Comment by LorisSigrist** [comment]
Makes sense, thanks for the reply. 
I'll try to reproduce the issue, your repo will be very helpful for doing that

**Comment by LorisSigrist** [comment]
Update:
I've identified where the bug happens, but I'm still confused about _why_ it happens.
The issue seems to be the `<svelte:element>` in your `<Button>` component. For some reason it renders out `<button href="undefined">` instead of just `<button>` when no `href` is given, and for some reason SvelteKit tries to crawl the `href` on the button. 

At least now that I know where it happens it should be easy to test & fix. 


> BTW, the `i18n.resolveRoute` should no longer be necessary on `<svelte:element>`, as the `href` there should also be automatically rewritten whenever the `<svelte:element>`  is an `<a>` tag. That's one of the new features in 0.3

<img width="635" alt="Screenshot 2024-01-30 at 11 58 20" src="https://github.com/opral/monorepo/assets/43482866/74a94e7e-ba5f-4f7e-ba19-c4578973b4f4">

**Comment by WarningImHack3r** [comment]
> but I'm still confused about _why_ it happens.

Average bug in software dev 😂

> it renders out `<button href="undefined">` instead of just `<button>` when no `href` is given

That's weird, idk if you mistyped it but the whole point of this component is to render an `<a href="x"` when there is an `href`, else render a `<button`. `<button href=*` shouldn't be a possible outcome.

> At least now that I know where it happens it should be easy to test & fix.

Great! Thanks for your efficiency!

> BTW, the `i18n.resolveRoute` should no longer be necessary on `<svelte:element>`, as the `href` there should also be automatically rewritten whenever the `<svelte:element>`  is an `<a>` tag. That's one of the new features in 0.3

Oh yeah I saw that one in the changelogs, I'll fix all that!  
I'm gonna open a PR that focuses on improving my usages (`deconstructPathname`), updating inlang deps separately from Renovate and using the new updates practices (`i18n.resolveRoute(href)`)

**Comment by WarningImHack3r** [comment]
Improvements in progress here: https://github.com/EmeraldHQ/Website/pull/210

**Comment by WarningImHack3r** [comment]
Thanks for fixing that @LorisSigrist!!
